### ISSUE TITLE
Training transformer model on dialydialog with poor results

### ISSUE THREAD

**Issue by bokanyibalazs7** [facebookresearch/ParlAI#1629]
Hi everyone,
after training a transfomer/generator model with the dailydialog task, I received poor result interacting with the trained model. Below you can find the parameters I used for training and a screenshot of the "conversation". 
My question is, what could be the cause of this? Should I set other parameters as well? Should I use the transformer/ranker model? Please let me know if you have any information regarding this topic.

The parameters from the command: 
1. -m transformer/generator
2. -t dailydialog
3. -bs 8
4. -eps 25
5. -esz 512
6. -nl 6
7. -hid 512
8. --dropout 0.1
9. --n-heads 8
10. -mf /user....

![image](https://user-images.githubusercontent.com/26432380/55953188-b7fb7480-5c5b-11e9-88a7-af53f6286a06.png)

**Comment by JohannesTK** [comment]
Seems like a problem with the interactive mode as I'm getting similar results with transfomer/generator. For me displaying the predictions works:

`python examples/display_model.py --model ${MODEL} -t ${TASK} -mf ${MODEL_FILE} -dt valid`

Something related to https://github.com/facebookresearch/ParlAI/pull/579?

**Comment by stephenroller** [comment]
I'm pretty sure you've just heavily overfit, and the messages you're entering into the model are sufficiently different from daily dialog for the model to be confused. Try copy pasting exact messages from the daily dialog validation set into the interactive session, and I bet it will do much well.

That transformer you're training is far too large for learning only daily dialog. You can make it slightly better by using pretrained embeddings (`-emb fasttext_cc`), but you're never really going to have success in this setup. You can make things more robust by multitasking it with a larger dataset (`-t twitter,dailydialog -et dailydialog`).

**Comment by stephenroller** [comment]
Actually, I've got an idea of what it is.

**Comment by ricsinaruto** [comment]
I also ran into this problem, and I don't think it's overfitting since I tried the model at the minimum validation loss that it reached.

**Comment by stephenroller** [comment]
Okay if you pull from master it should be able to do much better now without retraining.

**Comment by ricsinaruto** [comment]
Thanks @stephenroller for the quick fix, it's better now!
